SF-3454 Fix remaining issues with the new draft user interface

pmachapman opened the pull request:
This PR fixed the remaining issues with the new draft user interface. I have fixed each bug in a separate commit.

@Nateowami I am marking as a draft so you see if this fixes the issues you have been encountering, and if there are any additional bugs you are aware of to add to the  the ticket (https://jira.sil.org/browse/SF-3454)

Also, here is the interface for showing error details in a history item (the items will expand as data is available - this crash occurred before the build reached Serval):

![image](https://github.com/user-attachments/assets/112f961b-fa2b-4710-9d79-3026a97b4d3e)


<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/web-xforge/3297)
<!-- Reviewable:end -->

Nateowami commented:
@pmachapman I'm still seeing some issues. The testing I'm doing is generate one draft, then generate another draft, and making sure things make sense the entire time. 

Here are screenshots of the progression as I run the second draft:

"Draft is finishing". No it's not; it's starting.

![Screenshot from 2025-07-07 11-42-17](https://github.com/user-attachments/assets/07e85dc9-f6a1-4fef-b3bd-4636275a840f)

OK, now a few seconds later it correctly says it's queued.
![Screenshot from 2025-07-07 11-43-08](https://github.com/user-attachments/assets/becc2193-d4a6-4d5a-b573-0e54343bb17a)

Um, where did it go. It finished, but now it's not shown.
![Screenshot from 2025-07-07 11-47-39](https://github.com/user-attachments/assets/5a9ded13-9fa1-4a05-b7f8-30ecde7ee778)

I go to another page and come back, and now it's in a quantum superposition of being ready, and still finishing.
![Screenshot from 2025-07-07 11-47-54](https://github.com/user-attachments/assets/6ccc6c6f-e7d8-439c-b257-cdd83b529cfc)

Nateowami commented:
@pmachapman I'm using fast training, not echo. I can't be certain I restarted the server, so I'm trying it again, but having to wait for the queue.

Nateowami commented:
@pmachapman I re-tested this and can see some but not all of the issues using the echo engine. It seems like the problems are more pronounced with fast training.

pmachapman commented:
> It seems like the problems are more pronounced with fast training.

@Nateowami I have rerun tests with fast training enabled, but still cannot see the issues you are encountering. Let me know sometime we can do a shared session and look at it occurring on your computer.

pmachapman commented:
@Nateowami I have addressed some of the issues I saw on your machine. Let me know if this is any better...

pmachapman commented:
@Nateowami I have addressed some of the issues I saw on your machine. Let me know if this is any better... Also, any feedback you have on the screenshot I uploaded at the top of this PR RE: showing errors in the new draft interface.

Nateowami commented:
Excellent. In my testing it seems like the issues are now fixed.

pmachapman reviewed (commented):
> I was able to adjust it to look a little better by limiting the flex basis of the mat-panel-title with `flex: 0 1 50%` and adding some padding `padding: 8px 24px` to the mat-expansion-panel-header.

Done.

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/3297#-:-OV4JE0lDyN5mk2hxagD:br52jy4)* status: all files reviewed, 1 unresolved discussion (waiting on @RaymondLuong3)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-history-list/draft-history-entry/draft-history-entry.component.ts` line 207 at r2](https://reviewable.io/reviews/sillsdev/web-xforge/3297#-OUqYRIV65ETGkx_1L2-:-OV4JNi450lFoO3Jt7Hl:bu3jt7x) ([raw file](https://github.com/sillsdev/web-xforge/blob/1419e2e4a864538497fe184314d1dac05aebafc4/src/SIL.XForge.Scripture/ClientApp/src/app/translate/draft-generation/draft-history-list/draft-history-entry/draft-history-entry.component.ts#L207)):*
<details><summary><i>Previously, RaymondLuong3 (Raymond Luong) wrote…</i></summary><blockquote>

Nit: If we set this to be false to begin with, then the type of this can be a boolean.
</blockquote></details>

I allow it to be undefined so I can determine if it was set or not. This is because this component is reused for the latest draft (which will set `canDownloadBuild `) and historic draft versions (which will not set that value):
```ts
export class DraftHistoryEntryComponent {
  private _entry?: BuildDto;
  @Input() set entry(value: BuildDto | undefined) {
    this._entry = value;

    // Only set if a draft can be downloaded if it is not set externally
    if (this._canDownloadBuild == null) {
      this.canDownloadBuild = this._entry?.additionalInfo?.dateGenerated != null;
    }
```


<!-- Sent from Reviewable.io -->
